ctags.lua

```diff
@@ -129,11 +132,53 @@ local function get_match(word, path)
 	end
 end
 
-local function open_file(path, line)
-	vis:command(string.format('open %s', path))
+local function goto_tag(path, line)
+	local pos = {
+		path = vis.win.file.path,
+		line = vis.win.selection.line,
+		col  = vis.win.selection.col,
+	}
+
+	-- check path twice instead of testing vim:command() as it returns true even
+	-- if edit fails to open the file
+	if path ~= vis.win.file.path then
+		vis:command(string.format('e %s', path))
+	end
+	if path ~= vis.win.file.path then
+		return
+	end
```

Comment:
Wouldn't it be enough to check the `modified` flag of the current file before opening a new one to avoid the second `if`? I think it would be clearer and the comment would not be needed.

Reply:
Maybe. I'm looking at cmd_edit() to see if I can understand other times it could fail.